onnxruntime/contrib_ops/cpu/quantization/qlinear_softmax.cc

```diff
@@ -0,0 +1,371 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "contrib_ops/cpu/quantization/qlinear_softmax.h"
+
+#include <cstdint>
+#include <type_traits>
+#include <utility>
+
+#include "core/common/common.h"
+#include "core/providers/common.h"
+#include "core/providers/cpu/tensor/transpose.h"
+
+#include "core/mlas/inc/mlas.h"
+#include "core/platform/threadpool.h"
+#include "gsl/gsl-lite.hpp"
+
+
+namespace onnxruntime {
+namespace contrib {
+
+constexpr int OPSET13 = 13;
+
+namespace {
+template <typename T>
+void QlinearBuildLookupTableUint32(uint32_t* table,
+                                   const float x_scale,
+                                   size_t reduce_len) {
+  static_assert(std::is_same<T, int8_t>::value ||
+                    std::is_same<T, uint8_t>::value,
+                "unsupported quant type");
```

Comment:
you can add type constraint in the template itself

Reply:
Yes, I can. I am just thinking would that will comlicate build errors